code/modules/networks/computer3/bank_rec.dm

```diff
@@ -0,0 +1,1050 @@
+		"Genetics",
+		"Robotics",
+		"Medical",
+		"Cargo",
+		"Mining",
+		"Engineering",
+		"Catering",
+		"Hydroponics",
+		"Civilian",
+		"Research",
+		"Security",
+	)
+
+	var/static/list/budgets = list(
+		"Payroll",
+		"Shipping",
+		"Research",
+	)
+
+	var/static/list/team_to_job_datum = list(
+		"Stationwide" = list(),
+		"Genetics" = list(/datum/job/research/geneticist),
+		"Robotics" = list(/datum/job/research/roboticist),
+		"Cargo" = list(/datum/job/engineering/quartermaster, /datum/job/civilian/mail_courier),
+		"Mining" = list(/datum/job/engineering/miner),
+		"Engineering" = list(/datum/job/engineering/engineer, /datum/job/engineering/technical_assistant, /datum/job/command/chief_engineer),
+		"Research" = list(/datum/job/research/scientist, /datum/job/research/research_assistant, /datum/job/command/research_director),
+		"Catering" = list(/datum/job/civilian/chef, /datum/job/civilian/bartender, /datum/job/special/souschef, /datum/job/daily/waiter),
+		"Hydroponics" = list(/datum/job/civilian/botanist, /datum/job/civilian/rancher),
+		"Security" = list(/datum/job/security, /datum/job/command/head_of_security),
+		"Medical" = list(/datum/job/research/medical_doctor, /datum/job/research/medical_assistant, /datum/job/command/medical_director),
+		"Civilian" = list(/datum/job/civilian/janitor, /datum/job/civilian/chaplain, /datum/job/civilian/staff_assistant, /datum/job/civilian/clown,\
+		/datum/job/special), //Who really makes the world go round? At least one of these guys
+							//I can live with the sous chef getting paid in two categories
+							//If you have a special role and you're on the manifest everything is probably normal
+	)
+
+	// printer stuff
+	var/tmp/connected = 0
+	var/tmp/server_netid = null
+	var/tmp/potential_server_netid = null
+	var/tmp/selected_printer = null
+	var/tmp/list/known_printers = list()
+	var/tmp/printer_status = "???"
+
+	var/setup_logdump_name = "banklog"
+	var/setup_mail_freq = FREQ_PDA
+
+/datum/computer/file/terminal_program/bank_records/initialize()
+	if (..())
+		return TRUE
+
+	src.record_database = data_core.general
+	src.master.temp = null
+	src.menu = MENU_MAIN
+	src.field_input = 0
+
+	src.log_string += "<br><b>LOGIN:</b> [src.authenticated]"
+
+	src.print_text(src.mainmenu_text())
+
+/datum/computer/file/terminal_program/bank_records/input_text(text)
+	if(..())
+		return
+
+	var/list/command_list = parse_string(text)
+	var/command = command_list[1]
+	command_list -= command_list[1]
+
+	switch(menu)
+		if(MENU_MAIN)
+			switch(text2num_safe(command))
+				if(MENU_MAIN_OPT_QUIT)
+					src.print_text("Quitting...")
+					src.master.unload_program(src)
+					return
+				if(MENU_MAIN_OPT_INDEX)
+					src.record_database = data_core.general
+					src.menu = MENU_INDEX
+					src.print_index()
+				if(MENU_MAIN_OPT_SEARCH)
+					src.print_text("Please enter target name, ID, or rank.")
+					src.menu = MENU_SEARCH_INPUT
+					return
+				if(MENU_MAIN_OPT_BUDGET)
+					src.print_budget()
```

Comment:
Would be nice if this cleared the main menu from the top of the terminal like the other menu options do

Reply:
38e31c9e3d245710a7f79586f2b03ef339ece827